File: processing/SimpleSerialization/SwitchLED/LedStatus.java
/*
 * This is the LedStatus class whose instance (object) is going to be serialized
 * and sent via the serial connection.
 */

import jaron.simpleserialization.*;

public class LedStatus extends SerializationData {
  private int status = 0;

  public int getStatus() {
    return status;
  }

  public void setStatus(int status) {
    this.status = status;
  }

  public int getDataSize() {
    return SerializationTypes.SIZEOF_INTEGER;
  }

  public void readData(SerializationInputStream input) {
      status = input.readInteger();
  }

  public void writeData(SerializationOutputStream output) {
    output.writeInteger(status);
  }
}

File: arduino/examples/SwitchLED/LedStatus.cpp
/*
 * This is the definition of the LedStatus class whose instance (object) will
 * receive and deserialize the incomming serial data.
 */

#include "LedStatus.h"

int LedStatus::getDataSize() {
  return SerializationTypes::SIZEOF_INTEGER;
}

void LedStatus::readData(SerializationInputStream& input) {
  status = input.readInteger();
}

void LedStatus::writeData(SerializationOutputStream& output) {
  output.writeInteger(status);
}
